components/sup/src/main.rs

```diff
@@ -341,13 +342,13 @@ fn bldr_url_from_input(m: &ArgMatches) -> Option<String> {
 
 /// Resolve a channel. Taken from CLI args, or (failing that), a
 /// default value.
-fn channel(matches: &ArgMatches) -> String {
-    channel_from_input(matches).unwrap_or_else(channel::default)
+fn channel(matches: &ArgMatches) -> ChannelIdent {
+    channel_from_input(matches).unwrap_or_default()
```

Comment:
This feels like it wants to use `EnvConfig`

Reply:
Will add to the list for follow up